netx/netx.go

```diff
@@ -168,6 +175,15 @@ func NewDialer(config Config) Dialer {
 	return d
 }
 
+// NewHTTP3DNSDialer creates a new DNS Dialer for HTTP3 transport, with the resolver from the specified config
+func NewHTTP3DNSDialer(config Config) HTTP3Dialer {
```

Comment:
I would call this function `NewHTTP3Dialer` since the returned type is an `HTTP3Dialer`